src/App.tsx

```diff
@@ -1,203 +1,320 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, ChangeEvent, FormEvent } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { client } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>('all');
+  const [title, setTitle] = useState<string>('');
+  const [titleError, setTitleError] = useState<boolean>(false);
+  const [errorType, setErrorType] = useState<
+    null | 'load' | 'emptyTitle' | 'add' | 'delete' | 'update'
```

Comment:
you can make type for error and use here
type Error = 'load' | 'emptyTitle' | 'add' | 'delete' | 'update';
```suggestion
 const [errorType, setErrorType] = useState<Error | null>(null);
```

But in general type is redundant. Better to make enum for all possible error messages and keep message in state.

```
enum Error {
  NONE = '',
  LOAD_TODOS = 'Unable to load todos',
  EMPTY_TITLE = 'Title should not be empty',
  ADD_TODO = 'Unable to add a todo',
  DELETE_TODO = 'Unable to delete a todo',
  UPDATE_TODO = 'Unable to update a todo',
}